### ISSUE TITLE
[Bug]: [PersonaPicker] AutoFill behavior cannot be disabled - can create unexpected UX

### ISSUE THREAD

**Issue by marcosmoura** [microsoft/fluentui#23867]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
None
```

### Reproduction

https://aka.ms/fluentpen

### Bug Description

> Begin typing and press left or right arrow key when suggestions are off or not available
> Observe text field is cleared of user input

Expected
* Perhaps this expected case should only be if hosts opt in by disabling autofill - there should be a way to disable autofill at least
> Begin typing and press left or right arrow key when suggestions are off or not available
> Observe text field is not cleared and behaves how text normally would

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
